.gitlab-ci.yml

```diff
@@ -0,0 +1,203 @@
+variables: &variables
+  DEBIAN_FRONTEND: noninteractive
+  DOCKER_HOST: tcp://docker:2375
+  # Build parameters. We later override them at job level if needed.
+  SONAR_SCANNER: "OFF"
+  COVERALLS: "OFF"
+  RUN_TESTS: "ON"
+  STOCK_CPPZMQ: "ON"
+  USE_PCH: "ON"
+  CMAKE_BUILD_TYPE: "Debug"
+  TANGO_USE_USING_NAMESPACE: "ON"
+  BUILD_SHARED_LIBS: "ON"
+
+services:
+  - docker:20.10.1-dind
+
+.job-template: &job-template
+  image: debian:10
+  before_script:
+    - ulimit -c unlimited
+    - echo "core.%e.%p.%t" > /proc/sys/kernel/core_pattern
+    - apt-get update && apt-get install -y git wget unzip docker.io
+    - .travis/before_install.sh
+    - docker build -t cpp_tango .travis/${OS_TYPE}
+    - >
+        docker run
+        --rm
+        --name cpp_tango
+        --user root
+        -e BINTRAY_API_KEY=${CI_BINTRAY_API_KEY}
+        -e BINTRAY_USER_NAME=tango-ci
+        -e COVERALLS_REPO_TOKEN=${COVERALLS_REPO_TOKEN}
+        -e SONAR_TOKEN=${SONAR_TOKEN}
+        -e TANGO_HOST=${TANGO_HOST}
+        -e TRAVIS_BRANCH=${TRAVIS_BRANCH}
+        -e TRAVIS_JOB_ID=${TRAVIS_JOB_ID}
```

Comment:
Thanks for pointing those! I haven't noticed these variables when I copy-pasted the instructions. Since it is related to sonar, I propose to leave it as it is, so when we get back to sonar integration after migration we will be able to notice that something still needs to be changed here.

I'm all in for replacing sonar with something that we can run locally and that can produce html report similar to what we have in sonar. It is never good to depend on third-party, proprietary service.